Add demo for 2d inference on 3D volume

surajpaib opened the pull request:
Add demo for 2d inference on 3D volume

Signed-off-by: Suraj Pai <b.pai@maastrichtuniversity.nl>

part of #130 
https://github.com/Project-MONAI/MONAI/discussions/1641
https://github.com/Project-MONAI/MONAI/issues/3458

### Description
Tutorial that demonstrates how monai `SlidingWindowInferer` can be used when a 3D volume input needs to be provided slice-by-slice to a 2D model and finally, aggregated into a 3D volume.

### Status
**Ready**

wyli reviewed (commented):
thanks, would it be more generic if there's an option for the `slicing_axis`? so that the user can choose to squeeze to `[N, C, H, W]` or `[N, C, D, W]` or `[N, C, D, H]`

surajpaib commented:
@wyli Definitely. I'll make the edits soon

surajpaib commented:
Added a `slice_axis` argument which allows sliding over `D`, `H` and `W` axis.

wyli reviewed (approved):
Thanks, it looks good to me.

wyli commented:
thanks, there are some PEP format errors https://github.com/Project-MONAI/tutorials/runs/4504304679?check_suite_focus=true#step:7:130 otherwise we can merge it anytime

Nic-Ma commented on modules/2d_inference_3d_volume.ipynb:
```diff
@@ -0,0 +1,217 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #2.    !pip install monai
</pre><p>Could you please help change to:</p><pre class="ql-syntax" spellcheck="false">!python -c "import monai" || pip install -q "monai-weekly"
</pre><p>Refer to:</p><p>https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/unet_segmentation_3d_ignite.ipynb</p><p> </p><p>Thanks.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/479/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Nic-Ma replied:
<pre class="ql-syntax" spellcheck="false">Line #1.    from monai.inferers import SlidingWindowInferer
</pre><p>Maybe it's better to put all the <code>import</code>  in a notebook cell.</p><p> </p><p>Thanks.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/479/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Nic-Ma replied:
<p>Please the empty cell.</p><p> </p><p>Thanks.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/479/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Nic-Ma commented on modules/2d_inference_3d_volume.ipynb:
```diff
@@ -0,0 +1,217 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #46.                x = x.squeeze(dim=self.slice_axis + 2)
</pre><p>Maybe <code>self.slice_axis + 2</code> is not correct when <code>self.slice</code> is 2?</p><p> </p><p>Thanks.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/479/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

surajpaib replied:
It should still work when `self.slice_axis` is 2, this would mean it squeezes over the index 4 of the `[N,C,D,H,W]` tensor, which corresponds with `W`.

Nic-Ma replied:
Maybe I misunderstood the logic, I mean is there any case that `self.slice_axis + 2` out of index?

Thanks.

surajpaib replied:
Aah yes, I see.

So the `self.slice_axis` should only be used for axis 0, 1, 2 which corresponds to `[D, H, W]` respectively. I'll add a check to make this clear then!

Nic-Ma replied:
And I would suggest to use `spatial_dim` instead of `axis`, then users can easily understand the values.

Thanks.

surajpaib commented:
> 

Formatting should be PEP8 compliant now :)

surajpaib commented:
Added `assert` and changed `slice_axis` to `spatial_dim`. I decided to omit the word `slice` as I think that is evident given the context and the name of the class.

Nic-Ma reviewed (approved):
Thanks for the quick update, looks good to me!
